### DISCUSSION TITLE
Salesforce Refresh Token: https://medium.com/@bpmmendis94/obtain-access-refresh-tokens-from-salesforce-rest-api-a324fe4ccd9b

### DISCUSSION THREAD

**Discussion by g-r3m** [airbytehq/airbyte#37830] [Connector Questions] [answered]
Hello,

I'm trying to set up my first Salesforce connection with Airbyte. I try to set up a sandbox. 

I have tried to follow this article:[]( https://medium.com/@bpmmendis94/obtain-access-refresh-tokens-from-salesforce-rest-api-a324fe4ccd9b)
Here is the issue I have when I try to retrieve the refresh token _2. Get Access token & Refresh token_

I receive that the authorization code is expired.
<img width="476" alt="Screenshot 2024-05-04 at 20 12 44" src="https://github.com/airbytehq/airbyte/assets/35274967/100038a2-d409-4639-a8a8-97001f2dc22d">

So this lead to a few questions to me: 
- How do you specify that the instance is a sandbox in this article ?
  1 - During the creation of the app in Salesforce, Is there anything special in the callback url to set up ?
  2 - How to be sure when I try to GET the code from the sign-in that my code is valid, and for how long ?
  3 - Is there something else that I might have missed ?
  
 Thank you for your help, 
Regards,

**Comment by g-r3m** [comment]
Sorry the article is [missing](https://medium.com/@bpmmendis94/obtain-access-refresh-tokens-from-salesforce-rest-api-a324fe4ccd9b)

**Comment by g-r3m** [comment, marked as answer]
I found the solution. I had to change the instance URL.

**Comment by yanwari1** [comment]
Hi @g-r3m I am having the same problem as you, I am trying to connect to Sandbox but it is not working. Did you use your company's instance like the article says. My company's instance is deu14s but when i use this i get an error. What did you use for your return url?

https://<YOUR_INSTANCE>.salesforce.com/services/oauth2/authorize?response_type=code&client_id=<CONSUMER_KEY>&redirect_uri=https://login.salesforce.com/ 


error=redirect_uri_mismatch&error_description=redirect_uri%20must%20match%20configuration

Although the Return URL in salesforce App is similar to the URL for getting the code. 

could you please help me here? thank you
